src/Packetery/Module/Checkout.php

```diff
@@ -718,9 +739,28 @@ public function getShippingRates(): array {
 
 			$cost = $this->getRateCost( $options, $cartPrice, $cartWeight );
 			if ( null !== $cost ) {
-				$name                   = $this->getFormattedShippingMethodName( $options->getName(), $cost );
-				$rateId                 = ShippingMethod::PACKETERY_METHOD_ID . ':' . $optionId;
-				$customRates[ $rateId ] = $this->createShippingRate( $name, $rateId, $cost );
+				$name   = $this->getFormattedShippingMethodName( $options->getName(), $cost );
+				$rateId = ShippingMethod::PACKETERY_METHOD_ID . ':' . $optionId;
+				$taxes  = null;
+
+				if ( $cost > 0 && $this->options_provider->arePricesTaxInclusive() ) {
+					$rates            = WC_Tax::get_shipping_tax_rates();
+					$taxes            = WC_Tax::calc_inclusive_tax( $cost, $rates );
+					$taxExclusiveCost = $cost - array_sum( $taxes );
+					/**
+					 * Filters shipping taxes.
+					 *
+					 * @since 1.6.5
+					 *
+					 * @param array $taxes            Taxes.
+					 * @param float $taxExclusiveCost Tax exclusive cost.
+					 * @param array $rates            Rates.
+					 */
+					$taxes = apply_filters( 'woocommerce_calc_shipping_tax', $taxes, $taxExclusiveCost, $rates );
+					$cost -= array_sum( $taxes );
```

Comment:
Neměli bychom ošetřit situaci, kdy by se z filtru vrátilo něco jiného, než pole?